### PR TITLE
*: upgrade go1.23.2 | tidb-test=pr/2411 

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/50765

#55480

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by dveeden — APPROVED** [review]

**Comment by purelind** [comment]
/hold  

Waiting for the evaluation confirmation of @yudongusa and @bb7133 .

**Review by lance6716 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

File: pkg/expression/builtin_miscellaneous.go [hunk 1]
```diff
@@ -228,7 +228,7 @@ func (b *builtinLockSig) evalInt(ctx EvalContext, row chunk.Row) (int64, bool, e
 		return 0, false, err
 	}
 	if isNullTimeout {
-		timeout = 0 // Observed in MySQL, gets converted to 1s in TiDB because of min timeout.
+		timeout = 0 // Observed in MySQL, gets converted to 1s in TiDB because of minute timeout.
```
Comment (tangenta) [inline pkg/expression/builtin_miscellaneous.go:231]: I think it is not "minute"

Comment (hawkingrei) [reply pkg/expression/builtin_miscellaneous.go]: it has been fixed.

File: pkg/expression/builtin_info_test.go [hunk 2]
```diff
@@ -351,7 +351,7 @@ func TestFormatNanoTime(t *testing.T) {
 		{float64(2000), "2.00 us"},
 		{float64(898787877), "898.79 ms"},
 		{float64(9999999991), "10.00 s"},
-		{float64(898787877424), "14.98 min"},
+		{float64(898787877424), "14.98 minute"},
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply pkg/expression/builtin_info_test.go]: it has been fixed.

**Review by XuHuaiyu — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**